### ISSUE TITLE
Get a registered port number

### ISSUE THREAD

**Issue by bdarnell** [cockroachdb/cockroach#1868]
We should [request a registered port number from IANA](https://www.iana.org/form/ports-services) instead of just squatting on a port that looks unlikely to be used.

**Comment by spencerkimball** [comment]
Unsurprisingly, "cockroach" as a service name is not taken. Also, port 26257 is not taken, which is the result of typing out "cockr" on a phone keypad. Seems appropriate to request. In terms of all of the other items in the IANA application, here are the answers I propose:

- Assignee name: Cockroach Labs, Inc.
- Assignee email: spencer@cockroachlabs.com
- Contact name: Ben Darnell
- Contact email: ben@cockroachlabs.com
- Resources required: Port number and service name
- Transport protocols: TCP
- Service code: [blank]
- Service name: cockroach
- Desired port number: 26257
- Description: CockroachDB database listening for HTTP client and inter-node connections
- Reference: Proprietary protocol. Uses neither IP-layer broadcast, multicast, nor anycast communication.
- Defined TXT keys: [blank]
- If broadcast/multicast is used, how and what for?: Not used.
- If UDP is requested, please explain how traffic is limited, and whether the protocol reacts to congestion.: Not requested.
- If UDP is requested, please indicate whether the service is solely for the discovery of hosts supporting this protocol.: Not requested
- Please explain how your protocol supports versioning.: The protocol immediately initiates TLS (no STARTTLS). TLS version negotiation follows as usual.
- If your request is for more than one transport, please explain in detail how the protocol differs over each transport.: No other transport besides TCP requested.
- Please describe how your protocol supports security. Note that presently there is no IETF consensus on when it is appropriate to use a second port for an insecure version of a protocol.: The protocol immediately initiates TLS.
- Please explain why a unique port assignment is necessary as opposed to a port in range (49152-65535) or existing port.: Many CockroachDB server nodes are expected to be addressed through a load balancer or round-robin DNS by connecting clients and so require a well-defined port number. This is a new service application, so an existing port is not applicable.
- Please explain the state of development of your protocol.: The protocol is in use by beta deployments of CockroachDB servers.
- If SCTP is requested, is there an existing TCP and/or UDP service name or port number assignment? If yes, provide the existing service name and port number.: SCTP is not requested.
- What specific SCTP capability is used by the application such that a user who has the choice of both TCP (and/or UDP) and SCTP ports for this application would choose SCTP? See RFC 4960 section 7.1.: SCTP is not requested.
- Please provide any other information that would be helpful in understanding how this protocol differs from existing assigned services.: [blank]

**Comment by tamird** [comment]
On Thu, Jul 30, 2015 at 10:01 AM, Spencer Kimball <notifications@github.com>
wrote:

> CockroachDB database


Redundancy here, otherwise LG.

**Comment by bdarnell** [comment]
I've created a mailing list `iana-contact@` as the "assignee email" (with my name and address in the "contact" fields). 

Looking at existing entries, "description" is very short (e.g. "PostgreSQL Database"), so I'll just make that "CockroachDB". I also want to avoid mentioning HTTP so we don't get asked why we're not using 80/443. For versioning, I'll mention protocol buffers for versioning of the post-TLS protocol.

I've submitted the form; we can track the status at https://tools.iana.org/public-view/viewticket/837560
